### PR TITLE
Feature/DEV-13184: Select figure annotations using query paramters

### PR DESCRIPTION
**Pull request by hbalenda**
Changes:
- Adds `goToCanvasState` method to `javascript/application/canvas-panel.js` which scrolls to a figure, and can select annotations and/or a region
- Calls `goToCanvasState` on page load with data from query parameters

### CODE REVIEW

File: packages/11ty/content/_assets/javascript/application/canvas-panel.js [hunk 1]
```diff
@@ -1,51 +1,107 @@
...
+ * Handle annotation/choice selection with canvasPanel API
+ * {@link https://iiif-canvas-panel.netlify.app/docs/examples/handling-choice}
+ */
+const handleSelect = (element) => {
+  const figure = element.closest('.q-figure')
+  if (!figure) return
+  const canvasPanel = figure.querySelector('canvas-panel')
+  const annotationType = element.getAttribute('data-annotation-type')
+  const id = element.getAttribute('id')
+  const inputType = element.getAttribute('type')
```
Comment (mphstudios) [inline packages/11ty/content/_assets/javascript/application/canvas-panel.js:13]: Does it makes sense to destructure `element.attributes` for these values?

File: packages/11ty/content/_assets/javascript/application/canvas-panel.js [hunk 2]
```diff
@@ -1,51 +1,107 @@
...
+
+  const toggleChoice = ({ checked }) => {
+    canvasPanel.makeChoice(id, {
+      deselect: !checked,
+      deselectOthers: inputType === 'radio'
+    })
+    if (inputType !== 'checkbox') return
```
Comment (mphstudios) [inline packages/11ty/content/_assets/javascript/application/canvas-panel.js:20]: Whenever possible, the guard clause be the first statement, for example:
```suggestion
    if (inputType !== 'checkbox') return
    
    canvasPanel.makeChoice(id, {
      deselect: !checked,
      deselectOthers: inputType === 'radio'
    })
```

File: packages/11ty/content/_assets/javascript/application/canvas-panel.js [hunk 3]
```diff
@@ -1,51 +1,107 @@
...
-            canvasPanel.applyStyles(id, { opacity: Number(!!checked) })
-          }
-          toggleAnnotation(input)
-          input.addEventListener('click', ({ target }) => toggleAnnotation(target));
-          break
-        default:
-          console.warn(`Invalid annotation type ${annotationType}`)
-          break
-      }
+  if (region) canvasPanel.setAttribute('region', region)x
```
Comment (mphstudios) [inline packages/11ty/content/_assets/javascript/application/canvas-panel.js:61]: Is the `x` at the end of the statement a typo?
```suggestion
  if (region) canvasPanel.setAttribute('region', region)
```

File: packages/11ty/content/_assets/javascript/application/canvas-panel.js [hunk 4]
```diff
@@ -1,51 +1,107 @@
...
+  annotationIds.forEach((id) => {
+    const input = document.getElementById(id)
+    if (!input) {
+      console.warn(`Invalid annotation id: ${id}`)
```
Comment (mphstudios) [inline packages/11ty/content/_assets/javascript/application/canvas-panel.js:65]: Should we use our logger here to provide context in the console message?

File: packages/11ty/content/_assets/javascript/application/canvas-panel.js (hunk 2 above)
Comment (hbalenda) [reply packages/11ty/content/_assets/javascript/application/canvas-panel.js:21]: that guard clause is only guarding the things below it, maybe there is another way to write this though

File: packages/11ty/content/_assets/javascript/application/canvas-panel.js (hunk 1 above)
Comment (hbalenda) [reply packages/11ty/content/_assets/javascript/application/canvas-panel.js:14]: not sure how to do this in a way that is not very verbose, unless you have a suggestion?

**Review by mphstudios — APPROVED** [review]

**Pull request merged**